app/services/user_destroyer.rb

```diff
@@ -124,8 +124,9 @@ def block_external_urls(user)
   end
 
   def agree_with_flags(user)
+    guardian = Guardian.new(Discourse.system_user)
     ReviewableFlaggedPost.where(target_created_by: user).find_each do |reviewable|
-      reviewable.perform(@actor, :agree_and_keep)
+      reviewable.perform(@actor, :agree_and_keep) if reviewable.actions_for(guardian).has?(:agree_and_keep)
```

Comment:
Could this still fail if `@actor` doesn't have enough permissions, even if the system user can agree with the flag? Maybe we should check using `@guardian` instead?

Reply:
it's done.